internal/interpreter/dynamic_component.rs

```diff
@@ -884,7 +937,7 @@ pub(crate) fn generate_component<'id>(
             });
             self.item_array.push(unsafe { vtable::VOffset::from_raw(rt.vtable, offset) });
             self.original_elements.push(rc_item.clone());
-            debug_assert_eq!(self.original_elements.len(), self.tree_array.len());
+            // FIXME: debug_assert_eq!(self.original_elements.len(), self.tree_array.len());
```

Comment:
indeed, this is bad.  That's because you push tree_item for the embedding_item,  but not the original_elements.
This needs to be kept in sync i think.